common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -324,6 +325,29 @@ public  <T> T createPayload(String action, Map<String, String> parameters) {
         return (T) parameters.get("http_body");
     }
 
+    private Map<String,String> formatArrayParameters(Map<String, String> parameters) {
+        Map<String,String> newParameters =  new HashMap<>();
+        for (Map.Entry<String, String> entry : parameters.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            String escapedValue = escapeJsonArrayIfNeeded(value);
+            newParameters.put(key,escapedValue);
+        }
+        return newParameters;
+    }
+    protected String escapeJsonArrayIfNeeded(String value) {
+        if (isJsonArray(value)) {
+            return value.replaceAll("([^\\\\])\"", "$1\\\\\"");
+        }
+        return value;
+    }
+
+    protected boolean isJsonArray(String value) {
```

Comment:
Can you move this function to the Utils class?

Reply:
sure

Reply:
can we use [this method](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java#L98)?

Reply:
removed this method